packages/status-js/src/ethereum-client/ethereum-client.ts

```diff
@@ -42,4 +44,31 @@ export class EthereumClient {
       return
     }
   }
+
+  async resolveOwner(
+    registryContractAddress: string,
+    communityPublicKey: string
+  ): Promise<string | undefined> {
+    try {
+      const registryContract = new ethers.Contract(
+        registryContractAddress,
+        ['function getEntry(address _communityAddress) view returns (address)'],
+        this.#provider
+      )
+      const ownerContractAddress = await registryContract.getEntry(
+        publicKeyToETHAddress(communityPublicKey)
+      )
+
+      const ownerContract = new ethers.Contract(
+        ownerContractAddress,
+        ['function signerPublicKey() view returns (bytes)'],
+        this.#provider
+      )
+      const owner = await ownerContract.signerPublicKey()
```

Comment:
note: `% cast call --rpc-url https://optimism-mainnet.infura.io/v3/<INFURA_API_KEY> 0x9eDc11E5932372387E76ff3dcF66DB5465893823 "signerPublicKey()"         
0x0000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000004104953f5f0d355b37c39d1d6460a31ed1114455f8263b3fd1b84406c5f12c9eb7dfb76ba7513b92186010928254984fe98aee069b4c7e20f9ea3da497c3ae76947700000000000000000000000000000000000000000000000000000000000000`

Reply:
The result should match `message.signerPublicKey` to consider the waku payload valid.